universe/activities/activities.go

```diff
@@ -0,0 +1,249 @@
+package activities
+
+import (
+	"sort"
+
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+
+	"github.com/momentum-xyz/ubercontroller/database"
+	"github.com/momentum-xyz/ubercontroller/types"
+	"github.com/momentum-xyz/ubercontroller/types/entry"
+	"github.com/momentum-xyz/ubercontroller/types/generic"
+	"github.com/momentum-xyz/ubercontroller/universe"
+	"github.com/momentum-xyz/ubercontroller/universe/activity"
+	"github.com/momentum-xyz/ubercontroller/utils/umid"
+)
+
+var _ universe.Activities = (*Activities)(nil)
+
+type Activities struct {
+	ctx        types.NodeContext
+	log        *zap.SugaredLogger
+	db         database.DB
+	activities *generic.SyncMap[umid.UMID, universe.Activity]
+}
+
+func NewActivities(db database.DB) *Activities {
+	return &Activities{
+		db:         db,
+		activities: generic.NewSyncMap[umid.UMID, universe.Activity](0),
+	}
+}
+
+func (a *Activities) Initialize(ctx types.NodeContext) error {
+	a.ctx = ctx
+	a.log = ctx.Logger()
+
+	return nil
+}
+
+func (a *Activities) CreateActivity(activityID umid.UMID) (universe.Activity, error) {
+	activity := activity.NewActivity(activityID, a.db)
+
+	if err := activity.Initialize(a.ctx); err != nil {
+		return nil, errors.WithMessagef(err, "failed to initialize activity: %s", activityID)
+	}
+	if err := a.AddActivity(activity, false); err != nil {
+		return nil, errors.WithMessagef(err, "failed to add activity: %s", activityID)
+	}
+
+	return activity, nil
+}
+
+func (a *Activities) GetActivity(activityID umid.UMID) (universe.Activity, bool) {
+	asset, ok := a.activities.Load(activityID)
+	return asset, ok
+}
+
+func (a *Activities) GetActivities() map[umid.UMID]universe.Activity {
+	a.activities.Mu.RLock()
+	defer a.activities.Mu.RUnlock()
```

Comment:
Again, not a blocker, but SyncMap also implements it own locking, so it's kindof duplicate here.
(e.g. `a.activities.Map`)